examples/digital_fingerprinting/production/morpheus/dfp/stages/dfp_training.py

```diff
@@ -87,10 +89,42 @@ def supports_cpp_node(self):
 
     def accepted_types(self) -> typing.Tuple:
         """Indicate which input message types this stage accepts."""
-        return (MultiDFPMessage, )
+        return (
+            ControlMessage,
+            MultiDFPMessage,
+        )
 
+    def _dfp_mm_from_cm(self, control_message: ControlMessage) -> MultiDFPMessage:
+        """Create a MultiDFPMessage from a ControlMessage."""
+        user_id = control_message.get_metadata("user_id")
+        message_meta = control_message.payload()
```

Comment:
The general idea is metadata is a place to store arbitrary configuration items, while the payload is something common to every ControlMessage.

Another change coming with 23.11 is that the payload is going to be a more intelligent data storage object, one that's capable of dynamically caching its contents to make its use more or less independent of the size of its data. To start this will just be storage in system memory, or caching to disk, but also supporting iterating using torch datasets.